doc/source/cluster/kubernetes/examples/rayjob-kueue-priority-scheduling.md

```diff
@@ -0,0 +1,317 @@
+priority class.
+
+```yaml
+# kueue-resources.yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ResourceFlavor
+metadata:
+  name: "default-flavor"
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ClusterQueue
+metadata:
+  name: "cluster-queue"
+spec:
+  preemption:
+    reclaimWithinCohort: Any
+    withinClusterQueue: LowerPriority
+  namespaceSelector: {} # match all.
+  resourceGroups:
+  - coveredResources: ["cpu", "memory", "nvidia.com/gpu"]
+    flavors:
+    - name: "default-flavor"
+      resources:
+      - name: "cpu"
+        nominalQuota: 2
+      - name: "memory"
+        nominalQuota: 8G
+      - name: "nvidia.com/gpu"
+        nominalQuota: 1
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: LocalQueue
+metadata:
+  namespace: "default"
+  name: "user-queue"
+spec:
+  clusterQueue: "cluster-queue"
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: WorkloadPriorityClass
+metadata:
+  name: prod-priority
+value: 1000
+description: "Priority class for prod jobs"
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: WorkloadPriorityClass
+metadata:
+  name: dev-priority
+value: 100
+description: "Priority class for development jobs"
+```
+
+Create the Kueue resources:
+```bash
+kubectl apply -f kueue-resources.yaml
+```
+
+## Deploy a RayJob
+
+Download the RayJob that that will execute all the steps documented in [Fine-tune a PyTorch Lightning Text Classifier](https://docs.ray.io/en/master/train/examples/lightning/lightning_cola_advanced.html). The [source code](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/samples/pytorch-text-classifier) is also in the kuberay repo.
+
+```bash
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/pytorch-text-classifier/ray-job.pytorch-distributed-training.yaml
+```
+
+Before creating the RayJob, modify the RayJob metadata with:
+* a label to assign the RayJob to the LocalQueue created earlier.
+* a label to assign the RayJob with the `dev-priority` priority class.
+* Modify the name to indicate this is a job for development purposes.
+
+```yaml
+metadata:
+  generateName: dev-pytorch-text-classifier-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: dev-priority
+```
+
+Also note the resources required for this RayJob by looking at the resources requested by the nodes:
+```yaml
+resources:
+  limits:
+    cpu: "8"
+    memory: "20G"
+    nvidia.com/gpu: "1"
+  requests:
+    cpu: "8"
+    memory: "20G"
+    nvidia.com/gpu: "1"
+```
+
+Now deploy the RayJob:
+```bash
+$ kubectl create -f ray-job.pytorch-distributed-training.yaml
+rayjob.ray.io/dev-pytorch-text-classifier-r6d4p created
+```
+
+Verify that the RayCluster and Ray job are running:
+```bash
+$ kubectl get pod
+NAME                                                      READY   STATUS    RESTARTS   AGE
+dev-pytorch-text-classifier-r6d4p-4nczg                   1/1     Running   0          4s
```

Comment:
Kubernetes can detect 2 GPUs on the gpu-node-pool node, but the PyTorch can't.


```sh
kubectl get nodes "-o=custom-columns=NAME:.metadata.name,GPU:.status.allocatable.nvidia\.com/gpu"
# NAME                                                  GPU
# gke-kuberay-gpu-cluster-default-pool-775a3b37-qkm0    <none>
# gke-kuberay-gpu-cluster-gpu-node-pool-0375c869-f0mk   2
```

```python
# head Pod
import torch
print("Cuda support:", torch.cuda.is_available(),":", torch.cuda.device_count(), "devices")
```

Reply:
I tried to use `rayproject/ray-ml:2.9.0-cu118`, and got the same issue.

Reply:
There is no `nvidia-smi` binary.

Reply:
I identified the root cause. I removed `gpu-driver-version=latest` from the `gcloud container node-pools create gpu-node-pool` command and installed the GPU drivers via [Step 4: Install GPU drivers (optional)](https://anyscale-ray--42903.com.readthedocs.build/en/42903/cluster/kubernetes/user-guides/gcp-gke-gpu-cluster.html#step-4-install-gpu-drivers-optional). The `gpu-driver-version=latest` behavior seems to be a GKE bug.

Reply:
hmm... yeah you shouldn't need to install the GPU driver unless there was an issue. What GKE version were you using?

Reply:
I ran into this issue even without `gpu-driver-version=latest`. Weird because I didn't have GPU driver issues last week.

I'll file a bug report on this. For now I'll link to the steps to re-install the GPU driver if there's an issue

Reply:
@kevin85421 I tested this again with `gpu-driver-version=latest` and it worked. Some folks on the GKE team mentioned that  `gpu-driver-version=latest` is required for L4s at the moment. I added `gpu-driver-version=latest` back for now but let's revisit if you continue to see issues with L4 GPUs even with `gpu-driver-version=latest`.